src/hotspot/share/oops/markWord.hpp

```diff
@@ -44,6 +44,10 @@
 //  --------
 //  unused:25 hash:31 -->| unused_gap:1  age:4  unused_gap:1  lock:2 (normal object)
 //
+//  64 bits (with compact headers):
+//  -------------------------------
+//  nklass:32 hash:25 -->| unused_gap:1  age:4  self-fwded:1  lock:2 (normal object)
```

Comment:
```
  static const int unused_gap_bits                = LP64_ONLY(1) NOT_LP64(0);
  static const int age_bits                       = 4;
  static const int lock_bits                      = 2;
  static const int self_forwarded_bits            = 1;
```

We have `32 + 25 + 1 + 4 + 1 + 2 == 65`, so how are fitting all these fields in 64 bits?

Reply:
That's a mistake. The unused-gap should not be there with compact headers. I'll fix that.